src/common/transformations/src/transformations/transpose_sinking/ts_utils.cpp

```diff
@@ -418,6 +413,9 @@ bool HasSameOutputTransposeNodes(const NodePtr& main_node) {
                             transpose_axis_order.end(),
                             first_transpose_axis_order.begin()))
                 return false;
+            if (!is_sinking_node(input.get_node())) {
+                return false;
+            }
```

Comment:
It's a good idea to move is_sinking_node() call to one place. But now HasSameOutputTransposeNodes() does 2 roles:
- checking if a node has the same output transpose nodes
- checking if each transpose is sinking
Name of function is left the same. Later it will hard to understand why some node is not moving backward since there is no is_sinking_node() call in pattern check now. It is inside HasSameOutputTransposeNodes

Reply:
Consider to rename HasSameOutputTransposeNodes

Reply:
not so sure, maybe CheckTransposeConsumers?